content/en/tracing/connect_logs_and_traces/go.md

```diff
@@ -39,25 +39,19 @@ func handler(w http.ResponseWriter, r *http.Request) {
     span := tracer.StartSpan("web.request", tracer.ResourceName("/posts"))
     defer span.Finish()
 
-    // Retrieve Trace ID and Span ID
-    traceID := span.Context().TraceID()
-    spanID := span.Context().SpanID()
-
-    // Append them to log messages as fields:
-    log.Printf("my log message dd.trace_id=%d dd.span_id=%d", traceID, spanID)
+    // Append span info to log messages:
+    log.Printf("My log message %v", span)
```

Comment:
Shall we also provide the full format in case folks want to do it by hand?

Also we support JSON logs too right?

Reply:
I don't think we should encourage users to put the values in themselves. Not all of the values are exposed, and they need to be in a particular order, which the formatter takes care of.

 Is there a use case that that this doesn't cover?

We do support JSON logs, but that doesn't seem to be covered in this section for any language. Should we add that here, or to the other logging section? (https://docs-staging.datadoghq.com/knusbaum/update-go-tracer-logging/logs/log_collection/go/)

Reply:
I understand what you mean but I fear it's inconsistent with other docs.

For instance, our Go docs for general logs collection [recommend JSON logging](https://docs.datadoghq.com/logs/log_collection/go/):
> We strongly encourage setting up your logging library to produce your logs in JSON format

And you can see in [Python](https://docs.datadoghq.com/tracing/connect_logs_and_traces/python#manually-inject-trace-and-span-ids) and [Java](https://docs.datadoghq.com/tracing/connect_logs_and_traces/java?tab=log4j2) that we provide the raw format in case folks want to do it themselves. Maybe we need an overhaul in terms of how we present this info. Perhaps we can pitch it as more of an _advanced_ use case and we should lead with something easier/simpler. But either way we have a precedent for including this level of information.

cc @andrewsouthard1

Reply:
We can enable totally manual injection by exposing those values, but it will require more changes to the tracer. We can't do that today.

If this is important, we should prioritize it. For now, this API is available in `1.25.0`